src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java

```diff
@@ -0,0 +1,97 @@
+package de.plushnikov.intellij.plugin.inspection;
+
+import com.intellij.codeInsight.daemon.GroupNames;
+import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.*;
+import com.siyeh.ig.fixes.RemoveModifierFix;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static com.intellij.psi.PsiModifier.FINAL;
+import static com.intellij.psi.PsiModifier.PRIVATE;
+
+/**
+ * @author Plushnikov Michail
+ */
+public class RedundantModifiersOnValueLombokAnnotationInspection extends AbstractBaseJavaLocalInspectionTool {
+
+  @NotNull
+  @Override
+  public String getGroupDisplayName() {
+    return GroupNames.BUGS_GROUP_NAME;
+  }
+
+  @NotNull
+  @Override
+  public String getDisplayName() {
+    return "Redundant modifiers on @Value lombok annotations inspection";
+  }
+
+  @NotNull
+  @Override
+  public String getShortName() {
+    return "RedundantModifiersValueLombok";
+  }
+
+  @Override
+  public boolean isEnabledByDefault() {
+    return true;
+  }
+
+  @NotNull
+  @Override
+  public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder holder, final boolean isOnTheFly) {
+    return new LombokElementVisitor(holder);
+  }
+
+  private static class LombokElementVisitor extends JavaElementVisitor {
+
+    private final ProblemsHolder holder;
+
+    LombokElementVisitor(ProblemsHolder holder) {
+      this.holder = holder;
+    }
+
+    @Override
+    public void visitField(PsiField field) {
+      super.visitField(field);
+
+      final PsiElement parent = field.getParent();
+      if (parent instanceof PsiClass) {
+        final PsiClass clazz = (PsiClass) parent;
+        if (clazz.hasAnnotation("lombok.Value")) {
+          final PsiModifierList modifierList = field.getModifierList();
+          if (modifierList != null) {
+            if (modifierList.hasExplicitModifier(PRIVATE)) {
+              final Optional<PsiElement> psiPrivate = Arrays.stream(modifierList.getChildren())
+                .filter(psiElement -> psiElement.toString().contains("private"))
```

Comment:
Better compare PsiModifier constant with text of element:
PRIVATE.equals(psiElement.getText()) should be better

Reply:
good approach easy to refactor in future